components/brave_rewards/browser/database/publisher_info_database.cc

```diff
@@ -1132,6 +1129,52 @@ ledger::ServerPublisherInfoPtr PublisherInfoDatabase::GetServerPublisherInfo(
 
 // Other -------------------------------------------------------------------
 
+void PublisherInfoDatabase::RecordP3AStats(bool auto_contributions_on) {
+  if (!initialized_) {
+    return;
+  }
+
+  sql::Statement sql(
+      db_.GetCachedStatement(SQL_FROM_HERE,
+                             "SELECT category, COUNT(*) FROM contribution_info "
+                             "GROUP BY 1"));
+  int auto_contributions = 0;
+  int tips = 0;
+  int queued_recurring = 0;
+
+  while (sql.Step()) {
+    const int count = sql.ColumnInt(1);
+    switch (sql.ColumnInt(0)) {
+    case static_cast<int>(ledger::RewardsCategory::AUTO_CONTRIBUTE):
+      auto_contributions = count;
+      break;
+    case static_cast<int>(ledger::RewardsCategory::ONE_TIME_TIP):
+      tips += count;
+      break;
+    case static_cast<int>(ledger::RewardsCategory::RECURRING_TIP):
+      queued_recurring = count;
+      break;
+    default:
+      NOTREACHED();
+    }
+  }
+
+  if (queued_recurring == 0) {
+    // Check for queued recurring donations.
+    sql::Statement sql(
+        db_.GetUniqueStatement("SELECT COUNT(*) FROM recurring_donation"));
+    if (sql.Step()) {
+      queued_recurring = sql.ColumnInt(0);
+    }
+  }
+
+  const auto auto_contributions_state = auto_contributions_on ?
+        AutoContributionsP3AState::kAutoContributeOn :
+        AutoContributionsP3AState::kWalletCreatedAutoContributeOff;
+  RecordAutoContributionsState(auto_contributions_state, auto_contributions);
```

Comment:
do we actually need the value of `auto_contributions_state` here? Can't we independently record the state and the contribution count?

Reply:
we can't, see the function body